tbapy/main.py

```diff
@@ -27,10 +26,9 @@ def __init__(self, auth_key, auth_id=None, auth_secret=None, event_key=''):
         :param auth_secret: Your event authorization secret, obtainable at https://www.thebluealliance.com/request/apiwrite
         :param event_key: The event key that is linked to the ID and secret provided.
         """
-        self.auth_key = auth_key
-        self.auth_id = auth_id
         self.auth_secret = auth_secret
         self.event_key = event_key
+        self.session.headers.update({'X-TBA-Auth-Key': auth_key, 'X-TBA-Auth-Id': auth_id})
```

Comment:
If the user doesn't specify an auth_id because they only want to use the GET feature of the api, wouldn't it also include "None" (because they didn't set it) as the value of the 'X-TBA-Auth-Id' header for GET requests as well? I should probably check the TBA more to see this, but I just want to make sure because after running a request with a header of value None I get this:
`
requests.exceptions.InvalidHeader: Value for header {X-TBA-Auth-Id: None} must be of type str or bytes, not <class 'NoneType'>
`

Reply:
Other than this (which might not even be an issue) the code looks really good, and it looks like you're making good use of the requests library features. (There are a LOT, most of which we don't use)

Reply:
Weird, I thought the documentation(http://docs.python-requests.org/en/master/user/advanced/) for requests said I could use None to omit a header, but then again I didn't test for no input. I'll go ahead and update that.

Reply:
I believe what it meant on the docs is that you can omit the Session level headers (which you set with session.headers.update) by setting the same method level header to None (e.x. in a post request with the "headers" keyword argument)